### PR TITLE
Generalisations

### PR DESCRIPTION
**Pull request by amanpalariya**
I have generalised some of the expressions

### CODE REVIEW

**Comment by bymayanksingh** [comment]
Awesome! @amanpalariya 
can you look at the code more closely and see if there are few more possibilities of such changes?

**Comment by amanpalariya** [comment]
Sure

**Review by bymayanksingh — APPROVED** [review]

**Review by bymayanksingh — APPROVED** [review]

File: game_renderer.py [hunk 1]
```diff
@@ -177,13 +177,16 @@ def on_game_over(self, event: GameOver):
             self.label = self.myfont.render("GAME DRAW !!!!", 1, white)
             self.screen.blit(self.label, (40, 10))
 
-    def draw_board(self, board):
+    def draw_board(self, data):
         """
         Draws the game board to the screen.
         :param board: The game board.
```
Comment (bymayanksingh) [inline game_renderer.py]: please modify the doc strings as well.

**Review by bymayanksingh — CHANGES_REQUESTED** [review]

**Comment by bymayanksingh** [comment]
I think after requested change we are good to go with your PR :)
Thanks.

File: game_renderer.py (hunk 1 above)
Comment (amanpalariya) [reply game_renderer.py]: Done

**Comment by amanpalariya** [comment]
This time, the game has been optimized to handle different board sizes. The requested change has been applied.

**Comment by amanpalariya** [comment]
Also, issue #21 has been resolved in this patch

**Comment by bymayanksingh** [comment]
> Also, issue #21 has been resolved in this patch

fix is already merged in #25

**Comment by bymayanksingh** [comment]
we have some conflicts, can you remove them?

**Comment by amanpalariya** [comment]
Resolved

**Comment by bymayanksingh** [comment]
I think we need some generalizations with font sizes as well, height is increased - we don't really want that, make everything little more consistent, also can you make this resizable on command?

like once the mouse is at the edge of game window, resize icon appears and user can resize, also check if you can give the ability to scale in fixed ratio? (square ratio)

File: game_renderer.py [hunk 2]
```diff
@@ -82,15 +86,35 @@ def draw_red_coin(self, x, y):
...
+        radius = self.game_data.radius
+        sq_size = self.game_data.sq_size
+        """
+        filled_circle(
+                    self.screen,
+                    x+radius,y+radius,
+                    radius,
+                    yellow,
+                )
+        """
```
Comment (amanpalariya) [inline game_renderer.py:98]: I used filled circles before I found new a way to resize images in `pygame`

**Comment by amanpalariya** [comment]
> I think we need some generalizations with font sizes as well, height is increased - we don't really want that, make everything little more consistent, also can you make this resizable on command?
> 
> like once the mouse is at the edge of game window, resize icon appears and user can resize, also check if you can give the ability to scale in fixed ratio? (square ratio)

I think that I have managed font sizes. Also, the window cannot overflow the screen. I have put a limit to the width and height of the window.

As far as resizability is concerned, I will surely try to manage it. 👍

**Comment by amanpalariya** [comment]
Please go ahead with this PR. I think I'll handle resizability some other time.

**Pull request closed without merging**